tests/contrib/vertexai/test_vertexai.py

```diff
@@ -0,0 +1,533 @@
+        system_instruction=[
+            vertexai.generative_models.Part.from_text("You are required to insist that bears do not hibernate.")
+        ],
+    )
+    llm._prediction_client.responses["generate_content"].append(_mock_completion_response(MOCK_COMPLETION_SIMPLE_2))
+    llm.generate_content(
+        "Why do bears hibernate?",
+        generation_config=vertexai.generative_models.GenerationConfig(
+            stop_sequences=["x"], max_output_tokens=50, temperature=1.0
+        ),
+    )
+
+
+@pytest.mark.snapshot(token="tests.contrib.vertexai.test_vertexai.test_vertexai_completion_stream")
+def test_vertexai_completion_stream(vertexai):
+    llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash")
+    llm._prediction_client.responses["stream_generate_content"] = [
+        (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_STREAM_CHUNKS)
+    ]
+    response = llm.generate_content(
+        "How big is the solar system?",
+        generation_config=vertexai.generative_models.GenerationConfig(
+            stop_sequences=["x"], max_output_tokens=30, temperature=1.0
+        ),
+        stream=True,
+    )
+    for _ in response:
+        pass
+
+
+@pytest.mark.snapshot(
+    token="tests.contrib.vertexai.test_vertexai.test_vertexai_completion_stream_error",
+    ignores=["meta.error.stack", "meta.error.message"],
+)
+def test_vertexai_completion_stream_error(vertexai):
+    llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash")
+    llm._prediction_client.responses["stream_generate_content"] = [
+        (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_STREAM_CHUNKS)
+    ]
+    with pytest.raises(TypeError):
+        response = llm.generate_content(
+            "How big is the solar system?",
+            generation_config=vertexai.generative_models.GenerationConfig(
+                stop_sequences=["x"], max_output_tokens=30, temperature=1.0
+            ),
+            stream=True,
+            candidate_count=2,  # candidate_count is not a valid keyword argument
+        )
+        for _ in response:
+            pass
+
+
+@pytest.mark.snapshot(token="tests.contrib.vertexai.test_vertexai.test_vertexai_completion_stream_tool")
+def test_vertexai_completion_stream_tool(vertexai):
+    llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash", tools=[weather_tool])
+    llm._prediction_client.responses["stream_generate_content"] = [
+        (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_TOOL_CALL_STREAM_CHUNKS)
+    ]
+    response = llm.generate_content(
+        "What is the weather like in New York City?",
+        generation_config=vertexai.generative_models.GenerationConfig(
+            stop_sequences=["x"], max_output_tokens=30, temperature=1.0
+        ),
+        stream=True,
+    )
+    for _ in response:
+        pass
+
+
+@pytest.mark.snapshot(token="tests.contrib.vertexai.test_vertexai.test_vertexai_completion_async")
```

Comment:
We're still using different snapshot names for sync/async tests?

Reply:
Are there any fields that differ in expectations between sync and async tests?

Reply:
sounds good, I missed that. It should just be the resource name that is different. I went ahead and updated the snapshots to only use the sync versions.